src/transformers/training_args.py

```diff
@@ -807,6 +813,13 @@ def __post_init__(self):
                 )
             if not (self.sharded_ddp == "" or not self.sharded_ddp):
                 raise ValueError("sharded_ddp is not supported with bf16")
+
+        if self.adafactor:
+            warnings.warn(
+                "`adafactor` is deprecated and will be removed in version 5 of 🤗 Transformers. Use `--optim adafactor` instead",
+                FutureWarning,
+            )
```

Comment:
this is super-confusing to a user, there are 2 unrelated deprecations here.

1. the use of adafactor optimizer is deprecated
2. the use of `--adafactor` flag is deprecated over `--optim adafactor`

So these should be 2 different depredations. Does it make sense?

if my proposal to fix `self.optim = "adafactor"` is accepted then it should be possible.